docs/user-guide/glossary.md

```diff
@@ -34,6 +33,22 @@ counterexample
   They all refer to an assignment of values to all of the CVL variables and
   contract storage.  See {ref}`rule-overview`.
 
+overapproximation
+underapproximation
+  Sometimes it is useful to replace a complex piece of code with something
+  simpler that is easier to reason about.  If the approximation includes all of
+  the possible behaviors of the original code (and possibly others), it is
+  called an "overapproximation"; if it does not then it is called an
+  "underapproximation".  For example, a {ref}`NONDET <view-summary>` summary is
+  an overapproximation because every possible value that the original
+  implementation could return is considered by the Prover.
+
+  Proofs on overapproximated programs are {term}`sound`, but there may be
```

Comment:
![image](https://user-images.githubusercontent.com/38309115/230619369-a32cfdfb-a1b0-47a6-9f36-f3384314c380.png)

For those of us who benefit from visual aids in examples, would it make sense to include an annotated image somewhat like this one for the glossary entry on over- and under-approximation? In particular, it could be good to show which region might contain spurious (type I error) violations and which region might contain missed (type II error) violations.

Reply:
I feel like the glossary is maybe the wrong place for that much detail, although I'm not sure where the right place is (probably the user guide chapter on summarization?).  I'm also not sure this is the right image to include if we do decide to include one

Reply:
Definitely might be the wrong place for that detail, a user guide chapter on summarization could be good. I don't mean this image specifically, just something to visually show why over approximation could yield type I errors and underapproximation might yield type II errors. I initially thought that an example using sets would be good, but I didn't come up with anything that expressed the ideas well.